sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -48,7 +53,10 @@ object NestedColumnAliasing {
     case Project(projectList, child) =>
       Project(
         getNewProjectList(projectList, nestedFieldToAlias),
-        replaceChildrenWithAliases(child, attrToAliases))
+        replaceChildrenWithAliases(child, nestedFieldToAlias, attrToAliases))
+
+    case other =>
```

Comment:
This case only matches `Aggregate` and `Expand` now?

Reply:
Yea, `canPruneOn` decides what operator will be pruned on.

Reply:
Ah, I see. Thanks for the check. If so, it might be better to explicitly filiter them here for readablity like `case other if canPruneOn(plan)`, or leave some comments about that. Both is okay to me.

Reply:
Let me leave some comments, running `canPruneOn` twice seems redundant.